examples/inference-deployments/mpt/README.md

```diff
@@ -1,34 +1,130 @@
+
+## Sending requests to your deployment
+
+Once the deployment is ready, it's time to run inference! 
+
+<details>
+<summary> Using Python SDK </summary>
+
+
+```python
+from mcli import predict
+
+deployment = get_inference_deployment(<deployment-name>)
+input = {
     {
         "input_strings": "Write 3 reasons why you should train an AI model on domain specific data set.",
         "temperature": 0.01
     }
 }
+predict(deployment, input)
+
+```
+</details>
+
+<details>
+<summary> Using MCLI </summary>
+
+```bash
+mcli predict <deployment-name> --input '{"input_strings": ["hello world!"]}'
+
+```
+</details>
+
+<details>
+<summary> Using Curl </summary>
+
+```bash
+curl https://<deployment-name>.inf.hosted-on.mosaicml.hosting/predict_stream \
+-H "Authorization: <your_api_key>" \
+-d '{"input_strings": ["hello world!"]}'
 ```
+</details>
+
+<details>
+<summary> Using Langchain </summary>
+
+```python
+# Sign up for an account: https://forms.mosaicml.com/demo?utm_source=langchain
+
+from getpass import getpass
+
+MOSAICML_API_TOKEN = getpass()
+import os
 
-```{csv-table}
-:header: >
-:    "Parameters", "Type", "Required", "Default", "Description"
-:widths: 20, 20, 5, 10, 50
+os.environ["MOSAICML_API_TOKEN"] = MOSAICML_API_TOKEN
+from langchain.llms import MosaicML
+from langchain import PromptTemplate, LLMChain
+template = """Question: {question}"""
+
+prompt = PromptTemplate(template=template, input_variables=["question"])
+llm = MosaicML(inject_instruction_format=True, model_kwargs={'do_sample': False})
+llm_chain = LLMChain(prompt=prompt, llm=llm)
+question = "Write 3 reasons why you should train an AI model on domain specific data set."
+
+llm_chain.run(question)
 
-"input_string","List[str]","yes","N/A","The prompt to generate a completion for."
-"top_p","float","no","0.95","Defines the tokens that are within the sample operation of text generation. Add tokens in the sample for more probable to least probable until the sum of the probabilities is greater than top_p."
-"temperature","float","no","0.8","The temperature of the sampling operation. 1 means regular sampling, 0 means always take the highest score, 100.0 is getting closer to uniform probability."
-"max_length","int","no","256","Defines the maximum length in tokens of the output summary."
-"use_cache","bool","no","TRUE","Whether to use KV cacheing during autoregressive decoding. This will use more memory but improve speed."
-"do_sample","bool","no","TRUE","Whether or not to use sampling, use greedy decoding otherwise."
 ```
+</details>
+
+### Input parameters
+| Parameters | Type | Required | Default | Description |
+| --- | --- | --- | --- | --- |
+| input_string | List[str] | yes | N/A | The prompt to generate a completion for. |
+| top_p | float | no | 0.95 | Defines the tokens that are within the sample operation of text generation. Add tokens in the sample for more probable to least probable until the sum of the probabilities is greater than top_p |
```

Comment:
I think we support all HF generate kwargs here?

Reply:
Nope, there's a known issue where we accidentally convert all the floats to int, so removed them from this list. This is what we support at the moment, ideally we will fix that bug sometime to be able to support the rest of the kwargs lol

Reply:
ah ok